src/main/java/net/minecraftforge/event/world/jigsaw/ForgeVillagePools.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2019.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event.world.jigsaw;
+
+import net.minecraftforge.event.ForgeEventFactory;
+
+public class ForgeVillagePools {
+
+    public static void init(){
+    }
+
+    static {
+        ForgeEventFactory.onJigsawStructurePiecesInit(new JigsawStructurePoolInitEvent.Village());
```

Comment:
These special classes are a definite no, what was the logic behind this and the hacky static init event firing... fire the events normally at a specific time.

Reply:
because vanilla uses the same shitty way to add the Patters/Pieces only on time at runtime

Reply:
Just because vanilla does something dumb, doesnt mean we have to

Reply:
You already call init, so why don't you use that instead of this static initialiser?!?